Makefile

```diff
@@ -1,25 +1,37 @@
-TARGET        = phpbrew
-SIGNATURE     = $(TARGET).asc
+# See https://tech.davis-hansson.com/p/make/
+MAKEFLAGS += --warn-undefined-variables
+MAKEFLAGS += --no-builtin-rules
+
+PHPBREW_PHAR        = phpbrew
+SIGNATURE     = $(PHPBREW_PHAR).asc
 CP            = cp
 INSTALL_PATH  = /usr/local/bin
 TEST          = phpunit
 
-$(TARGET): vendor $(shell find bin/ shell/ src/ -type f) box.json.dist .git/HEAD
-	box compile
-	touch -c $@
+PHAR_SRC_FILES := $(shell find bin/ shell/ src/ -type f)
+
+
+.DEFAULT_GOAL := help
+
+
+.PHONY: help
+help:
+	@printf "\033[33mUsage:\033[0m\n  make TARGET\n\n\033[32m#\n# Commands\n#---------------------------------------------------------------------------\033[0m\n\n"
+	@fgrep -h "##" $(MAKEFILE_LIST) | fgrep -v fgrep | sed -e 's/\\$$//' | sed -e 's/##//' | awk 'BEGIN {FS = ":"}; {printf "\033[33m%s:\033[0m%s\n", $$1, $$2}'
 
-vendor: composer.lock
-	composer install
-	touch $@
 
-.PHONY: sign
-sign: $(SIGNATURE)
+.PHONY: build
+build:	## Builds PHPBrew PHAR
+build:
```

Comment:
pretty much the previous `make phpbrew`, with the difference that it will always compile the PHAR regardless of whether the PHAR was already built and up to date.

You also have `_build`, which I call "shadow command" and mimicks `build`, but there it does not care about the targets so it will nto rebuild the PHAR if not necessary.

I like to have this kind of command + shadow command to have a good UX whilst being able to keep shortucts of one needs it.

For example, in my projects I often have a `make serve` which will start the webserver, but will also boot up docker if not already booted up, install deps if not installed, migrate the DB if it is not up to date etc. It's a lot of stuff, and sometimes I just want to boot up the webserver without caring about all of that, and that's where I use a "shadow command", i.e. I would have `make _serve` that would just do that.

Note that here in this case, `_build` is effectively just an alias for `phpbrew` so might no be super useful. We can do without really